src/main/java/org/jabref/logic/importer/fetcher/DoiFetcher.java

```diff
@@ -81,7 +81,12 @@ public Optional<BibEntry> performSearchById(String identifier) throws FetcherExc
                     bibtexString = download.asString();
                 } catch (IOException e) {
                     // an IOException will be thrown if download is unable to download from the doiURL
-                    throw new FetcherException(Localization.lang("No DOI data exists"), e);
+                    // dispatch the IOException
+                    if (e instanceof FileNotFoundException) {
+                        throw new FetcherException(Localization.lang("Client Error"), e);
+                    } else {
```

Comment:
You don't seem to rely on actual http responses here. Is it robust, to just add an `else` without further checks here?

Reply:
I have found that the original http status check in `URLDownload.openConnection` won't throw an IOException if the http status is 4xx or 5xx. The original approach is to throw IOException int `URLDownload.openConnection` but it seem to fail fetcher test since it original shouldn't throw an exception. In my mind, I haven't come up with some good idea.

Reply:
Well, UrlDownload throws an IOException, so you could throw an IOException with the status code object, maybe and check it
The ideal solution would probably to have something like this... But this is probably overblown here
https://stackoverflow.com/a/47058425/3450689